apollo-runtime/src/main/java/com/apollographql/apollo/interceptor/ApolloAutoPersistedOperationInterceptor.java

```diff
@@ -108,4 +112,21 @@ boolean isPersistedQueryNotSupported(List<Error> errors) {
     }
     return false;
   }
+
+  public static class Factory implements ApolloInterceptorFactory {
+
+    final boolean useHttpGet;
```

Comment:
should we make it configurable to enable / disable autopersistence for queries / mutations?

Reply:
It's relatively easy to do so by implementing a custom `ApolloInterceptorFactory`:

```kotlin
class OnlyPersistQueriesInterceptor: ApolloInterceptorFactory {
    override fun newInterceptor(logger: Logger, operation: Operation<?, ?, ?>) = if (operation is Query) {
         ApolloAutoPersistedOperationInterceptor(logger, false);
    } else {
       null
    }
}
```

But it might be easier to add more params to the `Factory`. Something like this:

```java
Factory(boolean useHttpGetForQueries, boolean persistQueries, boolean persistMutations)
```
?

The `useHttpGetForQueries` feels a bit asymmetrical though. Maybe it is the one that should be left out instead?

Reply:
I was thinking of `Factory(boolean useHttpGetForQueries, boolean persistQueries, boolean persistMutations)`, so to keep everything in one place.

Reply:
Done. Let me know what you think.